### PR TITLE
feat(core): add PlainObject class that DTO's can extend to treat class as POJO

### PR DESCRIPTION
**Pull request by Langstra**

### CODE REVIEW

File: docs/docs/entity-manager.md [hunk 1]
```diff
@@ -137,7 +137,12 @@ const user1 = await orm.em.findOne(User, 1);
 
 As you can see in the fifth example, one can also use operators like `$and`, `$or`, `$gte`, 
 `$gt`, `$lte`, `$lt`, `$in`, `$nin`, `$eq`, `$ne`, `$like`, `$re`. More about that can be found in 
-[Query Conditions](query-conditions.md) section. 
+[Query Conditions](query-conditions.md) section.
+
+#### Using custom FilterQuery DTO
+If you decide to abstract the filter options in your own object then you might run into the problem that the find option does not return the results you'd expect. This is due to the fact that the FilterQuery should be provided as a POJO (Plain Old Json Object `{}`).
```
Comment (B4nan) [inline docs/docs/entity-manager.md:143]: POJO comes from java and means plain old java object, here i'd call it plain old js object, but calling it json object seems weird :]

File: docs/docs/entity-manager.md [hunk 2]
```diff
@@ -137,7 +137,12 @@ const user1 = await orm.em.findOne(User, 1);
...
+
+If you want to provide your own FilterQuery DTO, then your DTO class should extend the `PlainJson` class. This way MikroORM knows it should be treated as such.
```
Comment (B4nan) [inline docs/docs/entity-manager.md:145]: the class is called `PlainObject` but you are referring to it in docs as `PlainJson`. I like `PlainObject` better, so lets just fix the docs.

**Review by B4nan — CHANGES_REQUESTED** [review]
looking good, just two small docs issues

**Review by B4nan — APPROVED** [review]

**Comment by Langstra** [comment]
Can we merge this?

**Pull request merged**